lib/block_view_flush.go

```diff
@@ -948,9 +948,12 @@ func (bav *UtxoView) _flushDAOCoinBalanceEntriesToDbWithTxn(txn *badger.Txn, blo
 				balanceKey, computedBalanceKey)
 		}
 
+		// Either a zero entry or an isDeleted annotation results in a deletion.
+		isDeleted := balanceEntry.BalanceNanos.IsZero() || balanceEntry.isDeleted
```

Comment:
Ok, so I assume this is doing the following:

1. Before, if a balance went from non-zero to zero, we would let it linger in badger.
2. This change makes it so that we hard delete balances that hit zero.

Does that sound right? If yes, feel free to resolve this comment.

Reply:
This is actually a state-syncer specific bug. In particular, this addresses the issues we were seeing where max sells wouldn't disappear from postgres/graphql. Basically - when a balance entry hits zero, we have core logic that removes the entry entirely from badger. However, when we do this, we don't actually set `isDeleted` in core for some reason. In order for state syncer to know that this entry is going to be deleted, we need the above logic so that it can accurately mirror what's going on in badger.